intTests/test_mir_verify_slices/test.saw

```diff
@@ -0,0 +1,115 @@
+enable_experimental;
+
+let oneThroughFive = mir_term {{ [1, 2, 3, 4, 5] : [5][32] }};
+
+let f_spec_1 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_value a];
+
+  mir_return (mir_term {{ 3 : [32] }});
+};
+
+let f_spec_2 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 1 3];
+
+  mir_return (mir_term {{ 5 : [32] }});
+};
+
+// mir_slice_value must take an array reference as an argument.
+// Passing a bare array constitutes a type error.
+let f_fail_spec_1 = do {
+  let arr = mir_array_value mir_u32 [mir_term {{ 1 : [32] }}];
+  mir_execute_func [mir_slice_value arr];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
+
+// The end value of the range given to mir_slice_range_value must be less
+// than the length of the slice.
+let f_fail_spec_2 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 0 5];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
+
+// The start value of the range given to mir_slice_range_value must not
+// exceed the end value.
+let f_fail_spec_3 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 6 5];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
+
+// Indexing into a length-0 slice is disallowed.
+let f_fail_spec_4 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 0 0];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
+
+// f requires a slice of length at least two.
+let f_fail_spec_5 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 0 1];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
```

Comment:
Now I'm confused - does `mir_slice_range_value` take a half-open range or an inclusive one?  If it's inclusive, then `mir_slice_range_value a 0 1` should produce a slice of length 2 and this test should pass.

Reply:
It takes a half-open range (my earlier typo notwithstanding). Therefore, I believe this test case should remain in the expected-to-fail category—using the range `[0..1]` results in a slice of length 1, which causes `s[1]` to be out of bounds.